Library/brew.sh

```diff
@@ -140,12 +173,47 @@ case "$HOMEBREW_COMMAND" in
   --config)    HOMEBREW_COMMAND="config";;
 esac
 
-if [[ -f "$HOMEBREW_LIBRARY/Homebrew/cmd/$HOMEBREW_COMMAND.sh" ]] ; then
+if [[ -f "$HOMEBREW_LIBRARY/Homebrew/cmd/$HOMEBREW_COMMAND.sh" ]]
+then
   HOMEBREW_BASH_COMMAND="$HOMEBREW_LIBRARY/Homebrew/cmd/$HOMEBREW_COMMAND.sh"
-elif [[ -n "$HOMEBREW_DEVELOPER" && -f "$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOMEBREW_COMMAND.sh" ]] ; then
+elif [[ -n "$HOMEBREW_DEVELOPER" && -f "$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOMEBREW_COMMAND.sh" ]]
+then
   HOMEBREW_BASH_COMMAND="$HOMEBREW_LIBRARY/Homebrew/dev-cmd/$HOMEBREW_COMMAND.sh"
 fi
 
+if [[ "$HOMEBREW_COMMAND" = "help" || "$1" = "-h" || "$1" = "--help" ]]
+then
+
+  # Display sub command help for `brew help foo` and `brew foo --help`
+  if [[ "$HOMEBREW_ARG_COUNT" -gt 0 ]]
+  then
+    if [[ "$HOMEBREW_COMMAND" = "help" ]]
+    then
+      HOMEBREW_COMMAND="$HOMEBREW_LIBRARY/Homebrew/cmd/$1"
+    else
+      HOMEBREW_COMMAND="$HOMEBREW_LIBRARY/Homebrew/cmd/$HOMEBREW_COMMAND"
+    fi
+
+    if [[ -f "$HOMEBREW_COMMAND.sh" ]]
+    then
+      HOMEBREW_COMMAND="$HOMEBREW_COMMAND.sh"
+    elif [[ -f "$HOMEBREW_COMMAND.rb" ]]
+    then
+      HOMEBREW_COMMAND="$HOMEBREW_COMMAND.rb"
+    else
+      echo "Unknown command \`$@\`"
+      echo
+      echo "$HOMEBREW_HELP"
+      exit 1
+    fi
+    grep "#:" "$HOMEBREW_COMMAND" | cut -c3- | tr -d \` | sed "s/^\* //"
```

Comment:
Yes, the `^#:` sounds better, but I think it's going to be very hard to add back the backticks in the markdown. So I'd rather strip them out here, and keep the documentation comment markdown to make the man page easier.